webhook/main.go

```diff
@@ -0,0 +1,257 @@
+		return nil, fmt.Errorf("failed to setup app client: %w", err)
+	}
+	return app, nil
+}
+
+func getCloudBuildClient(ctx context.Context, logger *slog.Logger) (*cloudbuild.Client, error) {
+	client, err := cloudbuild.NewClient(ctx)
+	if err != nil {
+		logger.ErrorContext(ctx, "failed to create cloudbuild client", "error", err)
+		return nil, fmt.Errorf("failed to create cloudbuild client: %w", err)
+	}
+	return client, nil
+}
+
+func (s *Server) handler(resp http.ResponseWriter, req *http.Request) {
+	ctx := logging.WithLogger(req.Context(), s.logger)
+	payload, err := github.ValidatePayload(req, s.webhookSecret)
+	if err != nil {
+		s.logger.ErrorContext(ctx, "failed to validate payload", "error", err)
+		fmt.Fprint(resp, html.EscapeString("failed to validate payload"))
+		resp.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	event, err := github.ParseWebHook(github.WebHookType(req), payload)
+	if err != nil {
+		s.logger.ErrorContext(ctx, "failed to parse webhook", "error", err)
+		fmt.Fprint(resp, html.EscapeString("failed to parse webhook"))
+		resp.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	switch event := event.(type) {
+	case *github.WorkflowJobEvent:
+		if event.Action == nil || *event.Action != "queued" {
+			s.logger.InfoContext(ctx, "no action taken for action type", "action", *event.Action)
+			resp.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		if !slices.Contains(event.WorkflowJob.Labels, "self-hosted") {
+			s.logger.InfoContext(ctx, "no action taken for labels", "labels", event.WorkflowJob.Labels)
+			resp.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		installation, err := s.appClient.InstallationForID(ctx, strconv.FormatInt(*event.Installation.ID, 10))
+		if err != nil {
+			s.logger.ErrorContext(ctx, "failed to setup installation client", "error", err)
+			fmt.Fprint(resp, html.EscapeString("failed to setup installation client"))
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		httpClient := oauth2.NewClient(ctx, (*installation).AllReposOAuth2TokenSource(ctx, map[string]string{
+			"administration": "write",
+		}))
+		gh := github.NewClient(httpClient)
+		if s.baseURL != nil {
+			gh.BaseURL = s.baseURL
+		}
+
+		// Note that even though event.WorkflowJob.RunID is used for a dynamic string, it's not
+		// guaranteed that particular job will run on this specific runner.
+		jitconfig, _, err := gh.Actions.GenerateRepoJITConfig(ctx, *event.Org.Login, *event.Repo.Name, &github.GenerateJITConfigRequest{Name: fmt.Sprintf("GCP-%d", event.WorkflowJob.RunID), RunnerGroupID: 1, Labels: []string{"self-hosted", "Linux", "X64"}})
+		if err != nil {
+			s.logger.ErrorContext(ctx, "failed to generate jitconfig", "error", err)
+			fmt.Fprint(resp, html.EscapeString("failed to generate jitconfig"))
+			resp.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		projectID := os.Getenv("PROJECT_ID")
```

Comment:
nit; you should read these once at server startup, not inline during request processing.